### PR TITLE
Mongo

### PR DESCRIPTION
**Pull request by eliska-n**
Adding mongo init container. Compatible with Remote Control mongo technology MR

### CODE REVIEW

File: Site/ASAB Maestro/Files/mongo/replica.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Get hostnames from environment variable
+IFS=',' read -r -a hostnames <<< "$HOSTNAMES"
```
Comment (ateska) [inline Site/ASAB Maestro/Files/mongo/replica.sh:4]: `HOSTNAMES` is a bit specific.

I suggest `MONGO_HOSTNAMES`.

File: Site/ASAB Maestro/Files/mongo/replica.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	echo "Error: Failed to reconfigure the replica set."
+	break
+elif [[ $output == *"SUCCESS_RECONFIG"* ]]; then
+	echo "Success: Replica set reconfigured."
+	break
+elif [[ $output == *"SUCCESS_INITIATE"* ]]; then
+	echo "Success: New replica set initiated."
+	break
+fi
+done
```
Comment (ateska) [inline Site/ASAB Maestro/Files/mongo/replica.sh:39]: Newline.

File: Site/ASAB Maestro/Descriptors/mongo.yaml [hunk 3]
```diff
@@ -9,26 +9,19 @@ descriptor:
   volumes:
     - "{{SLOW_STORAGE}}/{{INSTANCE_ID}}/data:/data/db"
     - "{{SITE}}/{{INSTANCE_ID}}/conf:/etc/mongo:ro"
-    - "{{SITE}}/{{INSTANCE_ID}}/script:/script:ro"
+    - "{{SITE}}/{{INSTANCE_ID}}/mongo:/script:ro"
```
Comment (ateska) [inline Site/ASAB Maestro/Descriptors/mongo.yaml:12]: I recommend to keep this aligned: 
`"{{SITE}}/{{INSTANCE_ID}}/script:/script:ro"`

... to maintain at least bit of sanity ;-)

File: Site/ASAB Maestro/Descriptors/mongo.yaml [hunk 4]
```diff
@@ -9,26 +9,19 @@ descriptor:
...
 
   command: mongod --config /etc/mongo/mongod.conf
 
 files:
+# This is confusing - I can EITHER specify FOLDER in /Files/... dir OR a write a FILE directly, I was trying to specify unique files from /Files
```
Comment (ateska) [inline Site/ASAB Maestro/Descriptors/mongo.yaml:17]: I believe that both cases are supported ... and there is a documentation of that somewhere ;-)

**Review by ateska — CHANGES_REQUESTED** [review]

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 4 above)
Comment (eliska-n) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:17]: That's a bold statement

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 3 above)
Comment (ateska) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:12]: `{{INSTANCE_ID}}` is `mongo-1` e.g.

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 4 above)
Comment (ateska) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:17]: http://gitlab.teskalabs.int/asab/asab-remote-control/-/blob/master/remote_control/model/service.py#L390

File: Site/ASAB Maestro/Files/mongo/replica.sh [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+# Get hostnames from environment variable
+IFS=',' read -r -a hostnames <<< "$HOSTNAMES"
+
+port=27017
+
+# The JavaScript file to execute
+script=/script/reconfigureReplicaSet.js
+
+# Read the JSON file
+newConfig=$(cat /script/newConfig.json)
```
Comment (ateska) [inline Site/ASAB Maestro/Files/mongo/replica.sh:12]: `/script/new-config.json`

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 3 above)
Comment (eliska-n) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:12]: in this very repository

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 4 above)
Comment (eliska-n) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:17]: http://gitlab.teskalabs.int/asab/asab-remote-control/-/blob/master/remote_control/model/service.py#L488

Comment (ateska) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:17]: I see incorrect (old) comment.
Should I test this?

File: Site/ASAB Maestro/Descriptors/mongo.yaml (hunk 3 above)
Comment (ateska) [reply Site/ASAB Maestro/Descriptors/mongo.yaml:12]: Okej. I guess I got that ... but you don't need to put the script into the library.
You can build it directly to the "site" folder.

**Review by ateska — CHANGES_REQUESTED** [review]
Newline, then let's merge it.

**Pull request merged**